File: src/API/src/Modules/Products/Products.Application/DTOs/CreateProductRequest.cs
namespace Products.Application.DTOs
{
    public class CreateProductRequest
    {
        public string Code { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public Guid ProductTypeId { get; set; }
        public Guid DeploymentTypeId { get; set; }
        public Guid OwnershipTypeId { get; set; }
        public Guid? OwnerPartnerId { get; set; }
        public bool IsSubscriptionBased { get; set; }
        public bool IsLicenseBased { get; set; }
        public bool IsActive { get; set; } = true;
    }
}

File: src/ClientApp/src/app/pages/products/dtos/create-product.request.ts
export interface CreateProductRequest {
    code: string;
    name: string;
    productTypeId: string;
    deploymentTypeId: string;
    ownershipTypeId: string;
    ownerPartnerId?: string | null;
    description?: string | null;
    isSubscriptionBased: boolean;
    isLicenseBased: boolean;
    isActive: boolean;
}
